### DISCUSSION TITLE
Flow error when creating new allocations

### DISCUSSION THREAD

**Discussion by nhardenVenn** [jamessimone/apex-rollup#338] [Q&A]
Error Occurred:

An Apex error occurred: System.NullPointerException: Attempt to de-reference a null object
<img width="124" alt="Screen Shot 2022-06-23 at 11 16 09 AM" src="https://user-images.githubusercontent.com/107888703/175346638-3464c88e-aa37-4675-9e25-79a1937560d8.png">

How should I handle this error? I'm only seeing it when allocations are created. In this case we are summing up fund allocation amounts to the fund. 

Side question: Does apex-rollup play nice with NPSP?

Thank you,

Nate

**Comment by nhardenVenn** [comment]
<img width="1770" alt="Screen Shot 2022-06-23 at 11 15 43 AM" src="https://user-images.githubusercontent.com/107888703/175347551-fb906718-6879-491c-a5d9-cb95439359e4.png">

**Reply by jamessimone** [reply]
@nhardenVenn any chance you can keep your Apex Developer Console open while debugging the flow and report the full stacktrace for that error? I'll look into this on my end but at the moment, with the information provided, it's not clear where/how the error is occurring.

**Comment by jamessimone** [comment]
On the subject of NPSP - I have several non-profits using Apex Rollup in conjunction with NPSP, yes.

**Comment by nhardenVenn** [comment]
[apex-07L8Z00005Zu833UAB.log](https://github.com/jamessimone/apex-rollup/files/8969595/apex-07L8Z00005Zu833UAB.log)
I think this is what you're asking for...

**Comment by nhardenVenn** [comment]
<img width="729" alt="Screen Shot 2022-06-23 at 12 58 02 PM" src="https://user-images.githubusercontent.com/107888703/175364191-75c68ce7-5e1e-43db-b113-25f69ac1b1bc.png">

or this.

**Reply by jamessimone** [reply]
Thanks - either of the above was fine.

Since you have this flow rollup set to start from the parent records, you need to fill out the optional `Calc Item Type When Rollup Started From Parent` field on your rollup action.

**Reply by jamessimone** [reply]
I will be fixing this in the upcoming release, however, so that you don't have to provide that argument since it lives on the Rollup CMDT - that was just a miss on my part as far as putting this functionality together for the bulk invocable action. Apologies!

**Comment by nhardenVenn** [comment]
<img width="1194" alt="Screen Shot 2022-06-23 at 2 03 38 PM" src="https://user-images.githubusercontent.com/107888703/175378011-1f45bddb-034b-4d22-9331-15d9fda
<img width="839" alt="Screen Shot 2022-06-23 at 2 06 07 PM" src="https://user-images.githubusercontent.com/107888703/175378074-a01cfe56-c2b9-4121-8587-706808b77a62.png">
c090e.png">
<img width="834" alt="Screen Shot 2022-06-23 at 2 07 01 PM" src="https://user-images.githubusercontent.com/107888703/175378206-4ef25b09-a27b-4bec-8a41-304eb2912c8b.png">
<img width="817" alt="Screen Shot 2022-06-23 at 2 07 33 PM" src="https://user-images.githubusercontent.com/107888703/175378292-235d2138-9278-4bfb-ab79-d6c59cce298d.png">
<img width="1477" alt="Screen Shot 2022-06-23 at 2 08 03 PM" src="https://user-images.githubusercontent.com/107888703/175378398-26450cf6-5a69-4898-b28f-d91991e0d6c3.png">
<img width="1963" alt="Screen Shot 2022-06-23 at 2 11 18 PM" src="https://user-images.githubusercontent.com/107888703/175378861-430d4942-cacc-479c-aaae-e5af73592780.png">

Here is what I have at the moment. It doesn't like a filed I have in the where clause on my CMDT. What am I missing?

Thank you,

Nate

**Comment by nhardenVenn** [comment]
<img width="1194" alt="Screen Shot 2022-06-23 at 2 03 38 PM" src="https://user-images.githubusercontent.com/107888703/175379430-02c88e5c-9c81-40b6-84b2-1c4167d58110.png">
I left this out from above.

**Reply by jamessimone** [reply]
@nhardenVenn awesome, thanks for the screenshots - those really help! Is `Intacct_Sync_Status__c` a field on Fund Allocation?

**Comment by nhardenVenn** [comment]
yes.

**Reply by jamessimone** [reply]
Great. I'll have this issue wrapped up in just a bit. I don't know if you have seen my comments in #336 but there is current a block with packaging that will prevent me from releasing these changes - the current guidance is that packaging will be possible again late today, or tomorrow. I'll keep you apprised.

**Reply by jamessimone** [reply]
@nhardenVenn the new version just released should fix this for you!

**Reply by nhardenVenn** [reply]
This is great, thank you very much!

I do have a question, though.

I currently have it calculating from the parent object with the Recalculate Rollup Button, but is it possible to also allow the rollup calculation to trigger from the child simultaneously? Or do I have to choose one or the other?

Thank you,

Nate

**Reply by jamessimone** [reply]
@nhardenVenn yes, that is possible. as long as you don't actually have a flow/apex trigger calling Apex Rollup from the parent, and are only using the parent recalc button to qualify the CMDT records set up like that, then you can also have the children trigger the rollup with separate records!
